chroma-agent/chroma_agent/lib/node_admin.py

```diff
@@ -2,35 +2,39 @@
 # Use of this source code is governed by a MIT-style
 # license that can be found in the LICENSE file.
 
-
 from jinja2 import Environment, PackageLoader
-env = Environment(loader=PackageLoader('chroma_agent', 'templates'))
-
-from iml_common.lib.util import platform_info
 from chroma_agent.lib.shell import AgentShell
 
+env = Environment(loader=PackageLoader('chroma_agent', 'templates'))
+
 
 def write_ifcfg(device, mac_address, ipv4_address, ipv4_netmask):
     ifcfg_tmpl = env.get_template('ifcfg-nic')
     ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s" % device
     with open(ifcfg_path, "w") as f:
-        f.write(ifcfg_tmpl.render(device = device, mac_address = mac_address,
-                                  ipv4_address = ipv4_address,
-                                  ipv4_netmask = ipv4_netmask))
+        f.write(ifcfg_tmpl.render(device=device,
+                                  mac_address=mac_address,
+                                  ipv4_address=ipv4_address,
+                                  ipv4_netmask=ipv4_netmask))
 
     return ifcfg_path
 
 
 def unmanage_network(device, mac_address):
-    """Rewrite the network configuration file to set NM_CONTROLLED="no"
-    TODO: This is destructive and overwrites the file loosing all settings.
-    This needs to be fixed up.
+    """
+    Rewrite the network configuration file to set NM_CONTROLLED="no"
+
+    TODO: This is destructive and overwrites the file clearing
+    previously configured settings, needs fixing.
     """
     ifcfg_path = write_ifcfg(device, mac_address, None, None)
 
-    if platform_info.distro_version >= 7.0:
+    if ifcfg_path:
         try:
             AgentShell.try_run(['nmcli', 'con', 'load', ifcfg_path])
+        except OSError:
```

Comment:
_OSError_ seems overly broad here.  There could be all kinds of *OSError*s which are not _ENOENT_.

Reply:
any OSError is beyond the scope of this code to handle and indicates the relevant command is unavailable and should be ignored in this context

Reply:
I disagree.  Any failure to run the/a command, other than it not existing should not just ignore that error and carry on.  Perhaps we have no recovery path here, but in the least, the user should be alerted to the system error (so that they can resolve it) and we should stop trying to do what we are doing.

In this particular case, we are trying to tell NM not to manage the interface.  If we fail to do that, and continue on to try to use the interface ourselves, we are going to race and conflict with NM and get unpredictable results which will likely yield support tickets for us instead of clearly indicating to the user that they have a system problem that they need to resolve themselves.

Reply:
IMO causing an exception in the chroma-agent application and ceasing functionality due to the Network Manager being unavailable in the OS is a bad decision but in this case I will disagree and commit as I don't feel strongly enough about the issue to continue to contest